app/backend/src/couchers/servicers/threads.py

```diff
@@ -80,6 +93,8 @@ def GetThread(self, request, context):
                     )
                     for r in res[:page_size]
                 ]
+                all_res = session.query(Reply).filter(Reply.comment_id == database_id).all()
+                num_responses = len(all_res)
```

Comment:
You can just compute `func.count(Reply)` (I'm not sure if `func.count(Reply.id)` is faster or not?) to compute this in the db and not have to send all reply objects into the python backend.